### PR TITLE
Prevent most side effects of yanked modules

### PR DESCRIPTION
**Pull request by fmeum**
Yanked module versions no longer contribute dependency requirements or emit `DEBUG` messages for `print()` statements.

Since the module files of yanked modules are still evaluated to learn their compatibility levels, they can still fail to execute.

### CODE REVIEW

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by iancha1992** [comment]
@bazel-io fork 6.3.0

File: src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java [hunk 1]
```diff
@@ -291,6 +344,8 @@ private GetModuleFileResult getModuleFile(
         }
         result.moduleFile = moduleFile.get();
         result.registry = registry;
+        YankedVersionsUtil.getYankedInfo(registry, key, allowedYankedVersions, env.getListener())
```
Comment (Wyverald) [inline src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java:347]: nit -- this reads a bit awkwardly. maybe `result.yankedInfo = YankedVersionsUtil.getYankedInfo(...).orElse(null);` (or just use an Optional in the struct).

**Review by Wyverald — APPROVED** [review]

**Comment by iancha1992** [comment]
@Wyverald @fmeum Looks like in order to cherry-pick this to release-6.3.0, we'll need another commit/PR ("src/test/java/com/google/devtools/build/lib/rules/starlarkdocextract/StarlarkDocExtractTest.java" is missing). Could you please direct me to a commit/PR that helps? Thanks

**Comment by fmeum** [comment]
@iancha1992 Pretty sure that particular part of the diff can just be dropped as that file isn't part of 6.3.0.

**Comment by iancha1992** [comment]
@fmeum  Sorry there's also one more conflict from 

- "src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelModuleResolutionFunction.java" (conflicts caused by "compute" function which throws SkyFunctionException in the release-6.3.0 branch, but throws BazelModuleResolutionFunctionException exception in the master branch. Also "allowedYankedVersionsFromEnv" variable is no longer there in the master branch)

**Pull request closed without merging**